lib/prettify/index.ts

```diff
@@ -1,5 +1,8 @@
 export type Prettify<Type> = Type extends Function
   ? Type
-  : {
-      [Key in keyof Type]: Type[Key];
-    } & {};
```

Comment:
Removed the additional intersection with `{}`, don't think it's necessary.

Reply:
`& {}` is required to avoid seeing `Prettify<...>` in a more nested examples. See more in https://stackoverflow.com/a/78465372. So I'll suggest reverting this change

Reply:
The problem mentioned in https://stackoverflow.com/a/78465372 does happen when `Prettify` is _just_ a mapped type.

And it doesn't have to be nested; the problem shows up even with unions:

<img width="400" alt="image" src="https://github.com/user-attachments/assets/7ae8f7f4-a807-4f03-94e8-84d98c173c46">

<br><br>

However, our implementation of `Prettify` is _not_ merely a mapped type, there's a conditional that wraps the mapped type, so it doesn't suffer from the same problem. And the display tests ensure that this behaviour remains intact and doesn't break.

<img width="400" alt="image" src="https://github.com/user-attachments/assets/63f73ded-0510-4653-bbec-8500ed55ee8b">

<img width="400" alt="image" src="https://github.com/user-attachments/assets/b3cfdf45-0959-4b61-b419-fb1c8a7abf7e">

<br><br>

Playground link: https://tsplay.dev/WYxgQm

Reply:
Okay, I extended your checks - https://tsplay.dev/WKQoyN. It consists of (1) conditional type + mapped type, (2) mapped types with NO intersection and (3) mapped types WITH intersection. Only second solution resolves `Array<SimplePrettify<A>>` as `SimplePrettify<A>[]`

Reply:
I don't see any other reasons to keep `{}` so I'm happy to remove it for now